cli/internal/cache/cache_signature_authentication_test.go

```diff
@@ -93,8 +95,51 @@ func Test_SecretKeyErrors(t *testing.T) {
 	}
 }
 
+func generateRandomBytes() []byte {
+	length := rand.Intn(250)
+	b := make([]byte, length)
+	rand.Read(b)
+	return b
+}
+
+var letters = []rune("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ")
```

Comment:
I don't love the arbitrary nature of this restriction of runes we're considering as valid letters. Especially since it's missing 0-9 which will appear in a lot of hashes.

If we are only going to be fuzzing a subset of possible values of `string` I feel like `generateTag` should accept some newtype that validates at construction that the string only contains expected runes. I don't think we'd need to actually check this as I believe we're getting all of our hashes from `hex.Encode` which outputs `[0-f]`. I realize that requires quite a bit of plumbing, but it gives us a compile time guarantee that we're not passing in anything unexpected as `hash`.

Reply:
Yeah for hash I'm now creating a random sequence of bytes and hashing it to get our random hash